migration/20161007-cover-resolved-identifiers.py

```diff
@@ -25,16 +26,19 @@
 _db = production_session()
 source = DataSource.lookup(_db, DataSource.INTERNAL_PROCESSING)
 
+covered = _db.query(CoverageRecord.id).select_from(CoverageRecord).\
+    join(DataSource).filter(DataSource.name==source.name)
+covered = covered.subquery()
+
 # Get all of the Identifiers that can be resolved by the Metadata Wrangler
 # (ISBNs and Overdrive IDs only). Select only the ones that do not have
 # an either an UnresolvedIdentifier or a resolve-identifier CoverageRecord.
 resolved_identifiers = _db.query(Identifier).\
-    filter(Identifier.type.in_([Identifier.ISBN, Identifier.OVERDRIVE_ID])).\
-    outerjoin(Identifier.coverage_records).\
-    outerjoin(CoverageRecord.data_source).\
-    filter(or_(DataSource.name==None, DataSource.name!=source.name)).\
     outerjoin(Identifier.unresolved_identifier).\
-    filter(CoverageRecord.id==None, UnresolvedIdentifier.id==None).all()
+    filter(Identifier.type.in_([Identifier.ISBN, Identifier.OVERDRIVE_ID])).\
+    filter(UnresolvedIdentifier.id==None).\
+    filter(~Identifier.coverage_records.any(CoverageRecord.id.in_(covered))).\
+    options(lazyload(Identifier.licensed_through)).all()
```

Comment:
By default (set in `model.py`), LicensePool is eagerloaded with Identifiers; and LicensePools then eagerload DataSources. I'm resetting it not to load LicensePools-DataSources at all.

I don't think this was creating an issue before, but it was overcomplicating the SQL. Now we know for sure that the only DataSource in the mix is the CoverageRecord's.